docs/content/TextArea.mdx

```diff
@@ -0,0 +1,187 @@
+title: TextArea
+description: Use TextArea for multi-line text input form fields
+status: Alpha
+source: https://github.com/primer/react/blob/main/src/TextArea.tsx
+storybook: /react/storybook?path=/story/forms-textarea--default
+---
+
+import {TextArea} from '@primer/react'
+
+```js
+import {TextArea} from '@primer/react'
+```
+
+## Examples
+
+<Note variant="warning">
+
+TextArea components **must always** be accompanied by a corresponding label to improve support for assistive
+technologies. Examples below are provided for conciseness and may not reflect accessibility best practices.
+
+</Note>
+
+### Basic
+
+```javascript live noinline
+const TextAreaExample = () => {
+  // Running in controlled mode (recommended)
+  const [value, setValue] = React.useState('')
+
+  const handleChange = event => {
+    setValue(event.target.value)
+  }
+
+  return <TextArea onChange={handleChange}>{value}</TextArea>
+}
+
+render(<TextAreaExample />)
+```
+
+### Uncontrolled mode
+
+```javascript live noinline
+const TextAreaExample = () => {
+  const ref = React.useRef()
+
+  const handleSubmit = event => {
+    event.preventDefault()
+    if (!ref.current.value) {
+      alert(`Enter a value into the TextArea and press submit`)
+      return
+    }
+
+    alert(`Current TextArea value: ${ref.current.value}`)
+  }
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <TextArea ref={ref} defaultValue="Set the initial state in uncontrolled mode using the defaultValue prop" />
+      <br />
+      <Button type="submit">Submit</Button>
+    </form>
+  )
+}
+
+render(<TextAreaExample />)
+```
+
+### Displaying form validation state
+
+```jsx live
+<>
+  <Box as="label" for="success-state" sx={{display: 'block'}}>
+    Success state:
+  </Box>
+  <TextArea id="success-state" validationStatus="success" />
+  <Box as="label" for="error-state" sx={{display: 'block', mt: 5}}>
+    Error state:
+  </Box>
+  <TextArea id="error-state" validationStatus="error" />
+</>
+```
+
+### Inactive
+
+```jsx live
+<>
+  <TextArea disabled>This text is inactive</TextArea>
+</>
+```
+
+### Resizing and constraining dimensions
+
+By default, `TextArea` can be resized by the user vertically and horizontally.
+
+Resizing can be prevented by setting `resize` to `none`, with `height` and `width` props available for granular control of its appearance.
+Setting an explicit `height` or `width` can prevent its respective `resize` options from being applied.
+
+```jsx live
+<TextArea resize="none" height={400}></TextArea>
+```
+
+### Custom styling
+
+```jsx live
+<>
+  <TextArea sx={{marginBottom: 2}}></TextArea>
+  <p>Custom styles like `margin` and `padding` can be applied using the `sx` prop</p>
+</>
+```
+
+## Props
+
+### TextArea
+
+<PropsTable>
+  <PropsTableRow
+    defaultValue="undefined"
+    name="required"
+    type="boolean"
+    description="Indicates to the user and assistive technologies that the field value is required"
+  />
+  <PropsTableRow
+    name="block"
```

Comment:
I wonder if we should remove the `block` prop in favor of `sx={{width: '100%'}}` (in both TextArea and TextInput) cc @mperrotti

Reply:
I'd be down with that if that's all `block` is doing

Reply:
Feels like a very common use-case. Are we over-relying on `sx` here to handle scenarios many users would expect to be done via props?

Reply:
I'm okay with a prop for this use case if we think it's pretty common. I'm not the biggest fan of `block` as a name for this though. What do y'all think?

Reply:
I agree that `block` is a little unclear, but I'd prefer we change the naming for all components that use `block` if we choose to make it different here.